sirepo/package_data/static/js/cloudmc.js

```diff
@@ -3048,14 +3072,85 @@ SIREPO.viewLogic('tallySettingsView', function(appState, cloudmcService, panelSt
             'openmcAnimation.sourceNormalization',
             'openmcAnimation.thresholds',
         ], autoUpdate,
+        ['tallyReport.planePos'], updateEnergyPlot,
         ['openmcAnimation.tally'], validateTally,
         [
-            'tallyReport.planePos.numSteps',
             'tallyReport.selectedGeometry',
             'openmcAnimation.score',
             'openmcAnimation.showSources',
             'openmcAnimation.numSampleSourceParticles',
         ], showFields,
-        ['openmcAnimation.score'], preserveScore,
+        [
+            'openmcAnimation.tally',
+            'openmcAnimation.score',
+            'openmcAnimation.aspect',
+            'openmcAnimation.sourceNormalization',
+            'openmcAnimation.energyRangeSum',
+        ], invalidateThreshold,
     ];
+    $scope.$watch('tallyService.fieldData', showFields);
+});
+
+SIREPO.app.directive('energyRangeSlider', function(appState, cloudmcService, panelState, tallyService) {
+    return {
+        restrict: 'A',
+        scope: {
+            field: '<',
+            model: '=',
+        },
+        template: `
+          <div data-ng-if="steps">
+            <div data-slider="" data-model="model" data-field="field" data-min="min" data-max="max" data-steps="steps" data-is-multi="1" data-is-range="1" data-space="space"></div>
+          </div>
+        `,
+        controller: function($scope) {
+            $scope.tallyService = tallyService;
+
+            function updateRange() {
+                const f = cloudmcService.findFilter('energyFilter');                
+                if (f) {
+                    $scope.min = f.start;
+                    $scope.max = f.stop;
+                    $scope.steps = f.num;
+                    $scope.space = f.space;
+                }
+            }
+            updateRange();
+            $scope.$watch('tallyService.fieldData', (newValue, oldValue) => {
+                if (newValue !== oldValue) {
+                    updateRange();
+                }
+            });
+        },
+    };
+});
+
+SIREPO.app.directive('planePositionSlider', function(appState, panelState, tallyService) {
```

Comment:
Maybe a shared directive for energyRangeSlider and planePositionSlider? Seems like a lot of similarity